gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -134,7 +134,30 @@ private <T> T withRevWalkUnchecked(CheckedFunction1<RevWalk, T> fun) {
   }
 
   private boolean isBranchPresent(String branchFullName) {
-    return Try.of(() -> jgitRepoForMainGitDir.resolve(branchFullName)).getOrNull() != null;
+    // If '/' chars exist in the branch name, then this loop is needed in order to avoid possible IDE errors, which
+    // could appear if two files like in the example below exist under .git directory for the repository:
+    // .git/refs/heads/docs/improve_readme (short branch name is `docs/improve_readme` - exists locally)
+    // .git/refs/remotes/upstream/docs (short branch name is `docs` - this branch exists only on upstream, not locally)
+    // In such scenario `org.eclipse.jgit.lib.Repository#resolve` called to check if `docs/improve_readme` branch exists
+    // under .git/refs/remotes/upstream/ will try to find the branch using the following file path:
+    // .git/refs/remotes/upstream/docs/improve_readme
+    // which will end in a "Not a directory" `java.nio.file.FileSystemException` (as existing file
+    // .git/refs/remotes/upstream/docs is not a directory). And that will produce a LOG.error which will generate
```

Comment:
Pls refer to the specific method that calls `LOG.error`, also mention that the problem is with `org.slf4j.Logger#error(java.lang.String, java.lang.Throwable)`, not `org.slf4j.Logger#error(java.lang.String)`